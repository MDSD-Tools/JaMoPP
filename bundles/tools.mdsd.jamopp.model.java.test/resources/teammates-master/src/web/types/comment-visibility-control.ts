/**
 * The comment visibility controls.
 */
export enum CommentVisibilityControl {
  /**
   * Show comment visibility control.
   */
  SHOW_COMMENT = 'SHOW_COMMENT',

  /**
   * Show giver name visibility control.
   */
  SHOW_GIVER_NAME = 'SHOW_GIVER_NAME',
}
