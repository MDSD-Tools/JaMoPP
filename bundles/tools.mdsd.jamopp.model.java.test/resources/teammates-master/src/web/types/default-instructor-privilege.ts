import { DefaultInstructorPermissions } from './api-const';
import { InstructorPermissionSet } from './api-output';

/**
 * Structure for default co-owner privileges.
 */
export const DEFAULT_PRIVILEGE_COOWNER: () => InstructorPermissionSet =
    (): InstructorPermissionSet => {
      return JSON.parse(DefaultInstructorPermissions.COOWNER);
    };

/**
 * Structure for default manager privileges.
 */
export const DEFAULT_PRIVILEGE_MANAGER: () => InstructorPermissionSet =
    (): InstructorPermissionSet => {
      return JSON.parse(DefaultInstructorPermissions.MANAGER);
    };

/**
 * Structure for default observer privileges.
 */
export const DEFAULT_PRIVILEGE_OBSERVER: () => InstructorPermissionSet =
    (): InstructorPermissionSet => {
      return JSON.parse(DefaultInstructorPermissions.OBSERVER);
    };

/**
 * Structure for default tutor privileges.
 */
export const DEFAULT_PRIVILEGE_TUTOR: () => InstructorPermissionSet =
    (): InstructorPermissionSet => {
      return JSON.parse(DefaultInstructorPermissions.TUTOR);
    };

/**
 * The default instructor privilege.
 */
export const DEFAULT_INSTRUCTOR_PRIVILEGE: () => InstructorPermissionSet =
    (): InstructorPermissionSet => {
      return JSON.parse(DefaultInstructorPermissions.CUSTOM);
    };
