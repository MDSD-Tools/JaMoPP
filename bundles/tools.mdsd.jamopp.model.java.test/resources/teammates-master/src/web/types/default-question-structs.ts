import { QuestionTypeStructures } from './api-const';
import {
  FeedbackConstantSumQuestionDetails,
  FeedbackConstantSumResponseDetails,
  FeedbackContributionQuestionDetails,
  FeedbackContributionResponseDetails,
  FeedbackMcqQuestionDetails,
  FeedbackMcqResponseDetails,
  FeedbackMsqQuestionDetails,
  FeedbackMsqResponseDetails,
  FeedbackNumericalScaleQuestionDetails,
  FeedbackNumericalScaleResponseDetails,
  FeedbackRankOptionsQuestionDetails,
  FeedbackRankOptionsResponseDetails,
  FeedbackRankRecipientsQuestionDetails,
  FeedbackRankRecipientsResponseDetails,
  FeedbackRubricQuestionDetails,
  FeedbackRubricResponseDetails,
  FeedbackTextQuestionDetails,
  FeedbackTextResponseDetails,
} from './api-output';

/**
 * Structure for default text question details.
 */
export const DEFAULT_TEXT_QUESTION_DETAILS: () => FeedbackTextQuestionDetails =
    (): FeedbackTextQuestionDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_TEXT_QUESTION_DETAILS);
    };

/**
 * Structure for default text question response details.
 */
export const DEFAULT_TEXT_RESPONSE_DETAILS: () => FeedbackTextResponseDetails =
    (): FeedbackTextResponseDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_TEXT_RESPONSE_DETAILS);
    };

/**
 * Structure for default contribution question details.
 */
export const DEFAULT_CONTRIBUTION_QUESTION_DETAILS: () => FeedbackContributionQuestionDetails =
    (): FeedbackContributionQuestionDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_CONTRIBUTION_QUESTION_DETAILS);
    };

/**
 * Structure for default contribution question response details.
 */
export const DEFAULT_CONTRIBUTION_RESPONSE_DETAILS: () => FeedbackContributionResponseDetails =
    (): FeedbackContributionResponseDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_CONTRIBUTION_RESPONSE_DETAILS);
    };

/**
 * Structure for default constant sum (among options) question details.
 */
export const DEFAULT_CONSTSUM_OPTIONS_QUESTION_DETAILS: () => FeedbackConstantSumQuestionDetails =
    (): FeedbackConstantSumQuestionDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_CONSTSUM_OPTIONS_QUESTION_DETAILS);
    };

/**
 * Structure for default constant sum (among recipient) question details.
 */
export const DEFAULT_CONSTSUM_RECIPIENTS_QUESTION_DETAILS: () => FeedbackConstantSumQuestionDetails =
    (): FeedbackConstantSumQuestionDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_CONSTSUM_RECIPIENTS_QUESTION_DETAILS);
    };

/**
 * Structure for default constant sum question response details.
 */
export const DEFAULT_CONSTSUM_RESPONSE_DETAILS: () => FeedbackConstantSumResponseDetails =
    (): FeedbackConstantSumResponseDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_CONSTSUM_RESPONSE_DETAILS);
    };

/**
 * Structure for default numerical scale question details.
 */
export const DEFAULT_NUMSCALE_QUESTION_DETAILS: () => FeedbackNumericalScaleQuestionDetails =
    (): FeedbackNumericalScaleQuestionDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_NUMSCALE_QUESTION_DETAILS);
    };

/**
 * Structure for default numerical scale question response details.
 */
export const DEFAULT_NUMSCALE_RESPONSE_DETAILS: () => FeedbackNumericalScaleResponseDetails =
    (): FeedbackNumericalScaleResponseDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_NUMSCALE_RESPONSE_DETAILS);
    };

/**
 * Structure for default MCQ question details.
 */
export const DEFAULT_MCQ_QUESTION_DETAILS: () => FeedbackMcqQuestionDetails =
    (): FeedbackMcqQuestionDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_MCQ_QUESTION_DETAILS);
    };

/**
 * Structure for default MCQ question response details.
 */
export const DEFAULT_MCQ_RESPONSE_DETAILS: () => FeedbackMcqResponseDetails =
    (): FeedbackMcqResponseDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_MCQ_RESPONSE_DETAILS);
    };

/**
 * Structure for default MSQ question details.
 */
export const DEFAULT_MSQ_QUESTION_DETAILS: () => FeedbackMsqQuestionDetails =
    (): FeedbackMsqQuestionDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_MSQ_QUESTION_DETAILS);
    };

/**
 * Structure for default MSQ question response details.
 */
export const DEFAULT_MSQ_RESPONSE_DETAILS: () => FeedbackMsqResponseDetails =
    (): FeedbackMsqResponseDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_MSQ_RESPONSE_DETAILS);
    };

/**
 * Structure for default rubric question details.
 */
export const DEFAULT_RUBRIC_QUESTION_DETAILS: () => FeedbackRubricQuestionDetails =
    (): FeedbackRubricQuestionDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_RUBRIC_QUESTION_DETAILS);
    };

/**
 * Structure for default rubric question response details.
 */
export const DEFAULT_RUBRIC_RESPONSE_DETAILS: () => FeedbackRubricResponseDetails =
    (): FeedbackRubricResponseDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_RUBRIC_RESPONSE_DETAILS);
    };

/**
 * Structure for default rank options question details.
 */
export const DEFAULT_RANK_OPTIONS_QUESTION_DETAILS: () => FeedbackRankOptionsQuestionDetails =
    (): FeedbackRankOptionsQuestionDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_RANK_OPTIONS_QUESTION_DETAILS);
    };

/**
 * Structure for default rank options question response details.
 */
export const DEFAULT_RANK_OPTIONS_RESPONSE_DETAILS: () => FeedbackRankOptionsResponseDetails =
    (): FeedbackRankOptionsResponseDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_RANK_OPTIONS_RESPONSE_DETAILS);
    };

/**
 * Structure for default rank recipients question details.
 */
export const DEFAULT_RANK_RECIPIENTS_QUESTION_DETAILS: () => FeedbackRankRecipientsQuestionDetails =
    (): FeedbackRankRecipientsQuestionDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_RANK_RECIPIENTS_QUESTION_DETAILS);
    };

/**
 * Structure for default rank recipients question response details.
 */
export const DEFAULT_RANK_RECIPIENTS_RESPONSE_DETAILS: () => FeedbackRankRecipientsResponseDetails =
    (): FeedbackRankRecipientsResponseDetails => {
      return JSON.parse(QuestionTypeStructures.DEFAULT_RANK_RECIPIENTS_RESPONSE_DETAILS);
    };
