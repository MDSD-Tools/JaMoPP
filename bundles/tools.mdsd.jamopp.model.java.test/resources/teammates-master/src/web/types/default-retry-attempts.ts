/**
 * Default maximum number of times data loading can be retried
 */
export const DEFAULT_NUMBER_OF_RETRY_ATTEMPTS: number = 3;
