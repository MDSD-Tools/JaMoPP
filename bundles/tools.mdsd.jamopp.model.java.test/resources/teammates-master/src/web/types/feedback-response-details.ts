import { ApiConst } from './api-const';

/**
 * Special answer of a contribution question response to indicate the response is not submitted.
 */
export const CONTRIBUTION_POINT_NOT_SUBMITTED: number = ApiConst.CONTRIBUTION_POINT_NOT_SUBMITTED;

/**
 * Special answer of a contribution question response to indicate the response is not initialized.
 *
 * <p>Used in session result.
 */
export const CONTRIBUTION_POINT_NOT_INITIALIZED: number = ApiConst.CONTRIBUTION_POINT_NOT_INITIALIZED;

/**
 * Special answer of a contribution question response for 'Not Sure' answer.
 */
export const CONTRIBUTION_POINT_NOT_SURE: number = ApiConst.CONTRIBUTION_POINT_NOT_SURE;

/**
 * Special answer of a contribution question response for 'Equal Share' answer.
 */
export const CONTRIBUTION_POINT_EQUAL_SHARE: number = ApiConst.CONTRIBUTION_POINT_EQUAL_SHARE;

/**
 * Special answer of a MSQ question indicating 'None of the above'.
 */
export const MSQ_ANSWER_NONE_OF_THE_ABOVE: string = '';

/**
 * Special answer of a numerical scale question response to indicate the response is not submitted.
 */
export const NUMERICAL_SCALE_ANSWER_NOT_SUBMITTED: number = ApiConst.NUMERICAL_SCALE_ANSWER_NOT_SUBMITTED;

/**
 * Special answer of a rank options question response to indicate the response is not submitted.
 */
export const RANK_OPTIONS_ANSWER_NOT_SUBMITTED: number = ApiConst.RANK_OPTIONS_ANSWER_NOT_SUBMITTED;

/**
 * Special answer of a rank recipients question response to indicate the response is not submitted.
 */
export const RANK_RECIPIENTS_ANSWER_NOT_SUBMITTED: number = ApiConst.RANK_RECIPIENTS_ANSWER_NOT_SUBMITTED;

/**
 * Special answer of a rubric question response to indicate no choice is chosen.
 */
export const RUBRIC_ANSWER_NOT_CHOSEN: number = -1;

/**
 * Special value to indicate whether a value is present or not.
 */
export const NO_VALUE: number = ApiConst.NO_VALUE;
