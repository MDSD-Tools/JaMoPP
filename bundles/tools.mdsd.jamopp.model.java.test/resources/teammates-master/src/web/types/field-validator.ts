import { ApiConst } from './api-const';

/**
 * The maximum length of the feedback session name.
 */
export const FEEDBACK_SESSION_NAME_MAX_LENGTH: number = ApiConst.FEEDBACK_SESSION_NAME_MAX_LENGTH;

/**
 * The maximum length of the course id.
 */
export const COURSE_ID_MAX_LENGTH: number = ApiConst.COURSE_ID_MAX_LENGTH;

/**
 * The maximum length of the course name.
 */
export const COURSE_NAME_MAX_LENGTH: number = ApiConst.COURSE_NAME_MAX_LENGTH;
