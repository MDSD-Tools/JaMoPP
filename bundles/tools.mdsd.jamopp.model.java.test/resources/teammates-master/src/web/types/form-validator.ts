import { ApiConst } from './api-const';

/**
 * Represents the root FormValidator object of all form fields.
 */
export enum FormValidator {
  /**
   * Max length for the 'Course ID' field.
   */
  COURSE_ID_MAX_LENGTH = ApiConst.COURSE_ID_MAX_LENGTH,

  /**
   * Max length for the 'Course Name' field.
   */
  COURSE_NAME_MAX_LENGTH = ApiConst.COURSE_NAME_MAX_LENGTH,

  /**
   * Max length for the 'Student Name` field.
   */
  STUDENT_NAME_MAX_LENGTH = ApiConst.STUDENT_NAME_MAX_LENGTH,

  /**
   * Max length for the 'Section Name` field.
   */
  SECTION_NAME_MAX_LENGTH = ApiConst.SECTION_NAME_MAX_LENGTH,

  /**
   * Max length for the 'Team Name` field.
   */
  TEAM_NAME_MAX_LENGTH = ApiConst.TEAM_NAME_MAX_LENGTH,

  /**
   * Max length for the 'E-mail Address` field.
   */
  EMAIL_MAX_LENGTH = ApiConst.EMAIL_MAX_LENGTH,
}
