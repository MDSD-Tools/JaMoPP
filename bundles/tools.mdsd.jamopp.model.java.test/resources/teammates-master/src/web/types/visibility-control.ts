/**
 * The visibility controls.
 */
export enum VisibilityControl {

  /**
   * Show response visibility control.
   */
  SHOW_RESPONSE = 'SHOW_RESPONSE',

  /**
   * Show giver name visibility control.
   */
  SHOW_GIVER_NAME = 'SHOW_GIVER_NAME',

  /**
   * Show recipient name visibility control.
   */
  SHOW_RECIPIENT_NAME = 'SHOW_RECIPIENT_NAME',
}
