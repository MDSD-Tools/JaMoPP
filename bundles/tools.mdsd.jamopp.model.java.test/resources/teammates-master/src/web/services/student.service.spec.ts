import { TestBed } from '@angular/core/testing';

import { HttpClientTestingModule } from '@angular/common/http/testing';
import { of } from 'rxjs';
import { ResourceEndpoints } from '../types/api-const';
import { Course, Students } from '../types/api-output';
import { StudentUpdateRequest } from '../types/api-request';
import { CourseService } from './course.service';
import { HttpRequestService } from './http-request.service';
import { StudentService } from './student.service';
import DoneCallback = jest.DoneCallback;

const defaultStudentUpdateRequest: StudentUpdateRequest = {
  name: 'John Doe',
  email: 'johndoe@gmail.com',
  team: '',
  section: '',
  comments: '',
  isSessionSummarySendEmail: true,
};

const studentCsvListTester:
    (courseId: string, service: StudentService, spyCourseService: any, done: DoneCallback) => void =
    (courseId: string, service: StudentService, spyCourseService: any, done: DoneCallback): void => {
      const course: Course = require(`./test-data/${courseId}`).course;
      const students: Students = require(`./test-data/${courseId}`).students;
      spyOn(spyCourseService, 'getCourseAsInstructor').and.returnValue(of(course));
      spyOn(service, 'getStudentsFromCourse').and.returnValue(of(students));
      service.loadStudentListAsCsv({ courseId }).subscribe((csvResult: string) => {
        expect(csvResult).toMatchSnapshot();
        done();
      });
    };

describe('StudentService', () => {
  let spyHttpRequestService: any;
  let spyCourseService: any;
  let service: StudentService;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        HttpClientTestingModule,
      ],
      providers: [HttpRequestService, CourseService],
    });
    service = TestBed.inject(StudentService);
    spyHttpRequestService = TestBed.inject(HttpRequestService);
    spyCourseService = TestBed.inject(CourseService);
  });

  it('should be created', () => {
    expect(service).toBeTruthy();
  });

  it('should execute PUT when updating students in a course', () => {
    const paramMap: Record<string, string> = {
      courseid: 'CS3281',
      studentemail: 'johndoe@gmail.com',
    };
    spyOn(spyHttpRequestService, 'put').and.stub();

    service.updateStudent({
      courseId: paramMap.courseid,
      studentEmail: paramMap.studentemail,
      requestBody: defaultStudentUpdateRequest,
    });

    expect(spyHttpRequestService.put)
        .toHaveBeenCalledWith(ResourceEndpoints.STUDENT, paramMap, defaultStudentUpdateRequest);
  });

  it('should execute DELETE when deleting all students in a course', () => {
    const paramMap: Record<string, string> = {
      courseid: 'CS3281',
    };
    spyOn(spyHttpRequestService, 'delete').and.stub();

    service.deleteAllStudentsFromCourse({
      courseId: paramMap.courseid,
    });

    expect(spyHttpRequestService.delete)
        .toHaveBeenCalledWith(ResourceEndpoints.STUDENTS, paramMap);
  });

  it('should execute POST when regenerating key of a student in a course', () => {
    const paramMap: Record<string, string> = {
      courseid: 'CS3281',
      studentemail: 'johndoe@gmail.com',
    };
    spyOn(spyHttpRequestService, 'post').and.stub();

    service.regenerateStudentKey(paramMap.courseid, paramMap.studentemail);

    expect(spyHttpRequestService.post)
        .toHaveBeenCalledWith(ResourceEndpoints.STUDENT_KEY, paramMap);
  });

  it('should generate course student list with section as csv', (done: DoneCallback) => {
    studentCsvListTester('studentCsvListWithSection', service, spyCourseService, done);
  });

  it('should generate course student list without section as csv', (done: DoneCallback) => {
    studentCsvListTester('studentCsvListWithoutSection', service, spyCourseService, done);
  });
});
