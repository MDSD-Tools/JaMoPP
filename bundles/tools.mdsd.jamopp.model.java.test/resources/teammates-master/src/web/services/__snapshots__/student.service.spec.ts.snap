// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentService should generate course student list with section as csv 1`] = `
"Course ID,studentListWithSection
Course Name,Student List With Section

Section,Team,Name,Status,Email
Section A,N/A,student1OfNoTeam,Joined,student1OfNoTeam@test.com
Section A,Team 1,student1OfTeam1,Joined,student1OfTeam1@test.com
Section A,Team 1,student2OfTeam1,Joined,student2OfTeam1@test.com
Section A,Team 2,student1OfTeam2,Joined,student1OfTeam2@test.com
Section B,Team 1,student1OfTeam1,Joined,student1OfTeam1@test.com
Section B,Team 2,student1OfTeam2,Yet to Join,student1OfTeam2OfSectionB@test.com"
`;

exports[`StudentService should generate course student list without section as csv 1`] = `
"Course ID,studentListWithoutSection
Course Name,Student List Without Section

Team,Name,Status,Email
N/A,student1OfNoTeam,Joined,student1OfNoTeam@test.com
Team 1,student1OfTeam1,Joined,student1OfTeam1@test.com
Team 1,student2OfTeam1,Joined,student2OfTeam1@test.com
Team 2,student1OfTeam2,Joined,student1OfTeam2@test.com
Team 3,student1OfTeam3,Joined,student1OfTeam3@test.com
Team 3,student2OfTeam3,Yet to Join,student2OfTeam3@test.com"
`;
